src/System.Management.Automation/engine/CommandCompletion/CompletionCompleters.cs

```diff
@@ -4810,6 +4810,14 @@ private static List<CompletionResult> GetDefaultProviderResults(
                 bool hadErrors;
                 var childItemOutput = context.Helper.ExecuteCurrentPowerShell(out _, out hadErrors);
 
+                if (childItemOutput.Count == 1 &&
+                    (pathInfo.Provider.FullName + "::" + pathInfo.ProviderPath).EqualsOrdinalIgnoreCase(childItemOutput[0].Properties["PSPath"].Value as string))
+                {
+                    // Get-ChildItem returned the item itself instead of the children so there must be no child items to complete.
+                    // The filesystem provider does this when pointing to files: Get-ChildItem C:\pagefile.sys -Force and other providers may follow this pattern.
```

Comment:
As for this provider, I think the only thing worth mentioning is that this code doesn't work for it.

Reply:
I mean to remove the line:
```
// The filesystem provider does this when pointing to files: Get-ChildItem C:\pagefile.sys -Force and other providers may follow this pattern.
```